huxley/api/permissions.py

```diff
@@ -63,65 +62,94 @@ def has_permission(self, request, view):
         assignment = Assignment.objects.get(id=assignment_id)
         user = request.user
 
-        return (user.is_authenticated() and user.is_advisor() and
-                user.school.id == assignment.school.id)
+        return user_is_advisor(request, view, assignment.school_id)
 
 
-class IsSchoolDelegateAdvisorOrSuperuser(permissions.BasePermission):
-    '''Accept only the advisor of the given school with a given assignment.'''
+class AssignmentListPermission(permissions.BasePermission):
 
     def has_permission(self, request, view):
         if request.user.is_superuser:
             return True
 
-        delegate_id = view.kwargs.get('pk', None)
-        delegate = Delegate.objects.get(id=delegate_id)
-        user = request.user
-
-        return (user.is_authenticated() and user.is_advisor() and
-                user.school.id == delegate.school.id)
+        if request.method in permissions.SAFE_METHODS:
+            school_id = request.query_params.get('school_id', -1)
+            committee_id = request.query_params.get('committee_id', -1)
+            return (user_is_chair(request, view, committee_id) or 
+                    user_is_advisor(request, view, school_id))
 
+        return False
+        
 
-class AssignmentListPermission(permissions.BasePermission):
+class DelegateDetailPermission(permissions.BasePermission):
+    '''Accept requests to retrieve, update, and destroy a delegate from the
+       superuser and the advisor of the school of the delegate. Accept requests
+       to retrieve and update a delegate from the chair of the committee of 
+       the delegate.'''
 
     def has_permission(self, request, view):
-        if request.user.is_superuser:
+        user = request.user
+        if user.is_superuser:
             return True
 
-        if request.method in permissions.SAFE_METHODS:
-            return user_is_advisor(request, view)
+        delegate_id = view.kwargs.get('pk', None)
```

Comment:
I don't think we need this `try/accept`, the delegate ID is a required parameter of the DelegateDetail URI. We can just get away with saying
```python
delegate_id = view.kwargs['pk']
```